### PR TITLE
Added additional filters to report end point

### PR DESCRIPTION
**Pull request by sliverc**
Broken backwards compatibility as reports are now not per default filtered
per user anymore. A user filter need to set manually to gain the same
behavior.

### CODE REVIEW

**Comment by sliverc** [comment]
@czosel Security issue fixed that all users can edit all reports... Should be mergable now.

File: timed/tracking/permissions.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    """Changing an object is only allowed if object belongs to current user."""
+
+    def has_object_permission(self, request, view, obj):
+        if request.method in SAFE_METHODS:
+            return True
+
+        if obj.user_id == request.user.id:
+            return True
+
+        return False
```
Comment (czosel) [inline timed/tracking/permissions.py]: nitpick: `return obj.user_id == request.user.id`

Comment (sliverc) [inline timed/tracking/permissions.py]: I could do a one liner as well.. but I left this out for now... :) Adjusted.

**Comment by sliverc** [comment]
@czosel ready for review again

**Pull request merged**